bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/StackFrame.java

```diff
@@ -69,10 +70,17 @@
     // Indicate this frame belong to a transaction participant
     TransactionParticipantType trxParticipant;
 
+    public WDChannels parentChannels;
```

Comment:
shall we remove this? like now we can always peek the strand and get the parent channels from the previous stack frame. Earlier we did not have that as we were keeping channels in the strand, and strands don't have any other links, but with stack frames, you can always peek the previous frame.  The only place we need this is at the first stack frame(there is no parent frame, the parent is another strand)

Reply:
+1, will Fix.

Reply:
Still there is a issue. Now we handle channelsError like scenarios withis the stackframe. We can't do that if we remove thsi from stack since the stackframe stack is only accessible from the strand. I don't think moving back this logic to strand is clean. Correct me if I'm wrong.

Reply:
Yes you are correct, moving back to strand is not correct. What I'm saying is removing these hard links will clean up the code. Like if we can remove this, then there is a no hard link between stack frames, (we keep same channel reference in two frames) Do we have any scenarios where we only have the stack frame to get these channels?

Reply:
Fixed.